docs/ref-manual/cvl/methods.md

```diff
@@ -46,44 +55,280 @@ method_summary   ::= "ALWAYS" "(" value ")"
+the side effects of unknown methods.
+
+A useful middle ground is the `HAVOC_ECF` summary type.  A `HAVOC_ECF`
+summarization for a method encodes the assumption that the called method is not
+reentrant.  This summarization approximates a method call by assuming it can
+have arbitrary effects on contracts other than the contract being verified, but
+that it can neither change the current contract's state nor decrease its ETH
+balance (aside from value transferred by the method call itself).
+
+(dispatcher)=
+### `DISPATCHER` summaries
+
+The `DISPATCHER` summary type provides a useful approximation for methods of
+interfaces that are implemented by multiple contracts.  For example, the
+methods defined by the ERC20 specification are often summarized using the
+`DISPATCHER` summary type.
+
+If a function with a `DISPATCHER` summary is called, the Prover will assume
+that the receiver of the call is one of the known contract implementations
+containing the given signature; the call will then behave the same way that a
+normal method call on the receiver would.  The prover will consider examples
+with every possible implementing contract, but multiple `DISPATCHER` method
+calls on the same receiver address in the same example will use the same
+receiver contract.
+
+The set of contract implementations that the prover chooses from contains
+the set of contracts passed as [arguments to the CLI](../cli/options).
+In addition, the prover may consider an unknown target contract whose methods
+are all interpreted using the {ref}`AUTO summary <auto-summary>`.  The presence
+of the unknown contract is determined by the optional boolean argument to the
+`DISPATCHER` summary:
+
+ * With `DISPATCHER(false)` or just `DISPATCHER`, the unknown contract is
+   considered as a possiblity
+
+ * With `DISPATCHER(true)`, only the known contract instances are considered
 
-### `DISPATCHER`
+The most commonly used option is `DISPATCHER(true)`, because in most cases the
+behavior of `DISPATCHER(false)` is equivalent to that of `AUTO`.
+`DISPATCHER(false)` only behaves differently from `AUTO` when one of the known
+contracts exhibits behavior that `AUTO` assumes is impossible (such as
+reentrancy).
 
-### `AUTO`
+(auto-summary)=
+### `AUTO` summaries
 
-### expression and block summaries
+The behavior of the `AUTO` summary depends on the type of call[^opcodes]:
 
+ * Calls to non-library `view` and `pure` methods use the `NONDET` approximation:
+   they keep all state unchanged.
 
+ * Normal calls and constructors use the `HAVOC_ECF` approximation: they are
+   assumed to change the state of external contracts arbitrarily but to leave
+   the caller's state unchanged.
+
+ * Calls to library methods are assumed to change the caller's state in an
+   arbitrary way, but are assumed to leave the state of other contracts
+   unchanged.
+
+```{todo}
+The effect of library calls on the current contract's ETH balance is unclear.
```

Comment:
@jtoman

Reply:
external library calls are delegate calls, and by the yellow paper they cannot (and should not) transfer balances. Internal library calls are not message calls, and so do not directly change balances (unless the body of these functions have external, balance transferring calls)

Reply:
I replaced the third bullet by "Calls to library methods and `delegatecall`s are assumed to change the caller's storage in an arbitrary way, but are assumed to leave ETH balances and the storage of other contracts unchanged."  Is this accurate now?

Reply:
I think I've accurately updated the documentation